doc/rules-ios.md

```diff
@@ -1077,6 +1077,28 @@ simultaneously contain simulator and device architectures (that is, you can
 build a single framework artifact that works for all architectures by specifying
 `--ios_multi_cpus=i386,x86_64,armv7,arm64` when you build).
 
+`ios_static_framework` supports Swift, but there are some constraints:
+
+* `ios_static_framework` with Swift only works with Xcode 11 and above, since
+  the required Swift functionality for module compatibility is available in
+  Swift 5.1.
+* `ios_static_framework` only supports a single direct `swift_library` target
+  that does not depend transitively on any other `swift_library` targets. The
+  Swift compiler expects a framework to contain a single Swift module, and each
+  `swift_library` target is its own module by definition.
```

Comment:
Shouldn't this framework still be allowed to depend on other modules as long as those are also vendored as other static frameworks?

Reply:
But this is going to link everything it depends on, which can only be *_library targets, so how would it depend on one?

Reply:
Ah right I forgot this rule works this way. Internally our rule, that looks a lot like this one https://github.com/lyft/envoy-mobile/blob/master/bazel/swift_static_framework.bzl, doesn't fully link everything but does expect the consumer to have the dependent frameworks in their `deps`

Reply:
in theory we could do this with `avoid_deps` but support for that is not trivial as we'd need to subtract the correct transitive `swiftinterface` files from the provider.

Reply:
@sergiocampama Could you let us know what needs to be done to support `avoid_deps`? This limitation is kind of blocking us from updating to the new `rules_apple` version.